scripts/west_commands/sign.py

```diff
@@ -220,59 +234,118 @@ def sign(self, command, build_dir, bcfg, formats):
-                    "in build directory's ({}) .config".
-                    format(vtoff, build_dir))
-
-        if slot_size:
-            sign_base.extend(['--slot-size', str(slot_size)])
-        else:
-            log.wrn('expected nonzero slot size, but '
-                    'DT_FLASH_AREA_IMAGE_0_SIZE={} '
-                    "in build directory's ({}) device tree".
-                    format(slot_size, build_dir))
+        if size:
+            sign_base.extend(['--slot-size', str(size)])
+        sign_base.extend(args.tool_args)
 
-        b = pathlib.Path(build_dir)
-        cache = cmake.CMakeCache.from_build_dir(build_dir)
-        runner_config = cached_runner_config(build_dir, cache)
-
-        # Build a signed .bin
-        if 'bin' in formats and runner_config.bin_file:
+        log.banner('signed binaries:')
+        if in_bin:
             out_bin = args.sbin or str(b / 'zephyr' / 'zephyr.signed.bin')
-            log.inf('Generating:', out_bin)
-            sign_bin = (sign_base + args.tool_args +
-                        [runner_config.bin_file, out_bin])
+            sign_bin = sign_base + [in_bin, out_bin]
+            log.inf('bin: {}'.format(out_bin))
             log.dbg(quote_sh_list(sign_bin))
             subprocess.check_call(sign_bin)
-
-        # Build a signed .hex
-        if 'hex' in formats and runner_config.hex_file:
+        if in_hex:
             out_hex = args.shex or str(b / 'zephyr' / 'zephyr.signed.hex')
-            log.inf('Generating:', out_hex)
-            sign_hex = (sign_base + args.tool_args +
-                        [runner_config.hex_file, out_hex])
+            sign_hex = sign_base + [in_hex, out_hex]
+            log.inf('hex: {}'.format(out_hex))
             log.dbg(quote_sh_list(sign_hex))
             subprocess.check_call(sign_hex)
 
+    @staticmethod
+    def imgtool(command, args):
+        if args.tool_path:
+            command.check_force(shutil.which(args.tool_path),
+                                '--tool-path {}: not an executable'.
+                                format(args.tool_path))
+            tool_path = args.tool_path
+        else:
+            tool_path = shutil.which('imgtool')
+            if not tool_path:
+                log.die('imgtool not found; either install it',
+                        '(e.g. "pip3 install imgtool") or provide --tool-path')
+        return tool_path
+
     @staticmethod
     def get_cfg(command, bcfg, item):
         try:
             return bcfg[item]
         except KeyError:
             command.check_force(
-                False,
-                "imgtool parameter unknown, build directory has no {} {}".
-                format('device tree define' if item.startswith('DT_') else
-                       'Kconfig option',
-                       item))
+                False, "build .config is missing a {} value".format(item))
             return None
+
+    @staticmethod
+    def dt_params(b, cache):
```

Comment:
Wonder if this function could be called flash-something, since it seems to return information related to flash.

Reply:
It's not related to flash. It's retrieving parameters from devicetree that are needed to cryptographically sign an image. So it's called dt_params in a Signer class.

Reply:
Oh, you mean flash hardware. Sure, why not.